pkg/operation/botanist/namespaces.go

```diff
@@ -103,3 +119,28 @@ func (b *Botanist) WaitUntilSeedNamespaceDeleted(ctx context.Context) error {
 func (b *Botanist) DefaultShootNamespaces() component.DeployWaiter {
 	return namespaces.New(b.K8sSeedClient.Client(), b.Shoot.SeedNamespace)
 }
+
+// getShootRequiredExtensionTypes returns all extension types that are enabled or explicitly disabled for the shoot.
+// The function considers only extensions of kind `Extension`.
+func getShootRequiredExtensionTypes(ctx context.Context, b *Botanist) (sets.String, error) {
+	controllerRegistrationList := &gardencorev1beta1.ControllerRegistrationList{}
+	if err := b.K8sGardenClient.Client().List(ctx, controllerRegistrationList); err != nil {
+		return nil, err
+	}
+
+	requiredExtensions := shoot.ComputeRequiredExtensions(b.Shoot.GetInfo(), b.Seed.Info, controllerRegistrationList, b.Garden.InternalDomain, b.Shoot.ExternalDomain,
+		gardenletfeatures.FeatureGate.Enabled(features.UseDNSRecords))
+
+	types := sets.String{}
+	for kindType := range requiredExtensions {
+		split := strings.Split(kindType, "/")
+		if len(split) != 2 {
+			return nil, fmt.Errorf("unexpected required extension: %s", kindType)
+		}
+		if split[0] == extensionsv1alpha1.ExtensionResource {
+			types.Insert(split[1])
+		}
+	}
+
+	return types, nil
+}
```

Comment:
I rather vote to have the previous revision (without using the `ComputeRequiredExtensions` func). The current approach with reuse of `ComputeRequiredExtensions` seems "artifical" as we have to parse and filter the output and even pass some unrelated args to `ComputeRequiredExtensions` like a gardenlet feature gate.

Reply:
I see the point you made here and I reverted to a previous revision.